### PR TITLE
Fix bugs in `wcopy` and `update_mem_words`

### PR DESCRIPTION
**Pull request by wborgeaud**

### CODE REVIEW

File: evm/src/cpu/kernel/asm/memory/syscalls.asm [hunk 1]
```diff
@@ -124,14 +124,14 @@ sys_calldataload_after_mload_packing:
     // stack: kexit_info, dest_offset, offset, size
     DUP4 %num_bytes_to_num_words %mul_const(@GAS_COPY) %add_const(@GAS_VERYLOW) %charge_gas
 
-    %stack (kexit_info, dest_offset, offset, size) -> (dest_offset, size, dest_offset, offset, size, kexit_info)
+    %stack (kexit_info, dest_offset, offset, size) -> (dest_offset, size, kexit_info, dest_offset, offset, size)
     ADD // TODO: check for overflow, see discussion here https://github.com/mir-protocol/plonky2/pull/930/files/a4ea0965d79561c345e2f77836c07949c7e0bc69#r1143630253
     // stack: expanded_num_bytes, dest_offset, offset, size, kexit_info
```
Comment (dlubarov) [inline evm/src/cpu/kernel/asm/memory/syscalls.asm:129]: I think this comment is out of date?

File: evm/src/cpu/kernel/asm/memory/metadata.asm [hunk 2]
```diff
@@ -201,7 +201,7 @@ global sys_basefee:
     // stack: old_num_words, num_words, kexit_info
     DUP2 DUP2 GT
     // stack: old_num_words > num_words, old_num_words, num_words, kexit_info
-    %jumpi(%%end)
```
[the remark this answers is not in the document]
Comment (dlubarov) [reply evm/src/cpu/kernel/asm/memory/metadata.asm:204]: Ah good catch

**Review by dlubarov — APPROVED** [review]
LGTM

File: evm/src/cpu/kernel/asm/memory/syscalls.asm (hunk 1 above)
Comment (wborgeaud) [reply evm/src/cpu/kernel/asm/memory/syscalls.asm]: Yes it is. Fixed

**Pull request merged**